src/containers/upcoming-events/ducks/thunks.ts

```diff
@@ -1,11 +1,18 @@
-import { EventInformation, EventsThunkAction } from './types';
+import { getPrivilegeLevel } from '../../../auth/ducks/selectors';
+import { PrivilegeLevel } from '../../../auth/ducks/types';
 import { upcomingEvents } from './actions';
+import { EventInformation, EventsThunkAction } from './types';
 
 export const getUpcomingEvents = (): EventsThunkAction<void> => {
-  return (dispatch, getState, { publicApiClient }) => {
+  return (dispatch, getState, { publicApiClient, protectedApiClient }) => {
+    const eventsFetcher: () => Promise<EventInformation[]> =
+      getPrivilegeLevel(getState().authenticationState.tokens) ===
+      PrivilegeLevel.NONE
+        ? publicApiClient.getUpcomingEvents
+        : protectedApiClient.getProtectedEvents;
+
     dispatch(upcomingEvents.loading());
-    return publicApiClient
-      .getUpcomingEvents()
+    return eventsFetcher()
```

Comment:
⛏️ Using a noun for a function name is generally bad, we would rather use a verb, i.e. fetchEvents()